uniffi_bindgen/src/interface/mod.rs

```diff
@@ -911,33 +921,119 @@ impl APIConverter<Field> for weedle::dictionary::DictionaryMember<'_> {
         if let Type::Object(_) = type_ {
             bail!("Objects cannot currently appear in record fields");
         }
+        let default = match self.default {
+            None => None,
+            Some(v) => Some(v.value.convert(ci, &type_)?),
+        };
         Ok(Field {
             name: self.identifier.0.to_string(),
             type_,
             required: self.required.is_some(),
-            default: match self.default {
-                None => None,
-                Some(v) => Some(v.value.convert(ci)?),
-            },
+            default,
         })
     }
 }
 
+#[derive(Debug, Clone, Copy, Serialize, Deserialize)]
+pub enum Radix {
+    Decimal = 10,
+    Octal = 8,
+    Hexadecimal = 16,
+}
+
 // Represents a literal value.
 // Used for e.g. default argument values.
 #[derive(Debug, Clone, Serialize, Deserialize)]
 pub enum Literal {
     Boolean(bool),
     String(String),
-    // TODO: more types of literal
+    UInt(u64, Radix, Type),
+    Int(i64, Radix, Type),
+    Float(f64, Type),
```

Comment:
I'm looking for feedback on the internal representation here.

We infer the type of the number from the type of the argument.

Initially I wanted to use the corresponding rust type to be represented, but that meant an awful lot of duplication, especially around radixes and `format!("{}", n)`.

The simplest thing to do would be to use to pass the `String` straight to the bindings, and then let the bindings figure out the differences in parsing and rendering numbers between IDL and their own language; e.g. `e` v `E`, octal llteral. This is not perfect, but this might yet be the right solution.

Presented here is a halfway between these two: using 8 byte numbers to represent signed and unsigned integers, and floats. We have to do a little bit of extra work— mostly involving the regex crate and some extra `format!` calls— but not much; it also brings the small but real possibility of accuracy lost converting between `f64` and `f32`, though I'm not sure how to quantify that.

Reply:
This PR needs rebasing, and I notice `f64` needs to derive `Hash`. The simplest thing to do here is to move `Literal::Float` to use the weedle parsed `String`.